### ISSUE TITLE
Type 2 SCD with datetime column

### ISSUE THREAD

**Issue by iiLaurens** [chrthomsen/pygrametl#40]
For type 2 SCD tables, when the toatt and fromatt columns are datetime objects (so YYYY-mm-dd HH:MM:ss), the time  ignored when checking for equality of the srcdateatt and fromatt of the previous row. This could lead to different versions being created even if the srcdateatt and fromatt were the same.

Specifically, I am referring to [line 1157 to 1160](https://github.com/chrthomsen/pygrametl/blob/master/pygrametl/tables.py#L1157).

Shouldn't the logic include for the possibility of having datetime columns?

**Comment by iiLaurens** [comment]
You are right this only occurs in the fallback comparison. After further inspection I noticed there is a deeper issue that puts me in this 'fallback comparison' section of the codebase. The deeper issue is that return types from the `other` object representing the previous line can return inconsistent types. This happened for me when when using a sqlite3 database with a datetime column for the `toatt` and `fromatt` columns. Instead or return datetime types it would return string objects (or vice versa, I don't remember exactly anymore). This caused the `toatt` values for the  `row` and `other` objects to be of different types, pushing me to the fallback comparison you talked about.

Trying to get the types for `toatt` and `fromatt` consistent wasn't actually easy for sqlite3. One issue I ran into is that sometimes a datetime typed object is passed to the default srcdateparser, which expects a string. I solved this by creating a custom srcdateparser like this:

```python
def robust_datetime_parser(ymdhmsstr):
    if isinstance(ymdhmsstr, datetime.datetime):
        return ymdhmsstr  # Already a datetime object
    return pygrametl.ymdhmsparser(ymdhmsstr)```

Ideally you would do some strict typechecking (and perhaps type hinting so that users of this library know what kind of types to provide) to prevent type mismatches going into the library and coming out from the variety of database engines. But including the little patch to the default srcdateparser as seen above helps solve some of the annoying symptoms when we are already dealing with datetime objects and not with strings.
